core/loadpoint.go

```diff
@@ -1303,17 +1304,33 @@ func (lp *Loadpoint) publishChargeProgress() {
 
 // socPollAllowed validates charging state against polling mode
 func (lp *Loadpoint) socPollAllowed() bool {
+	if lp.charging() {
+		// always update Soc when charging
+		lp.didChargeOnLastSocUpdate = true
+		return true
+	}
+
+	if lp.connected() && lp.socUpdated.IsZero() {
```

Comment:
this will poll forever if soc update keeps returning errors

Reply:
But then this was also the case previously.

Reply:
Also if the soc update returns an error, `socUpdated` is still set:
https://github.com/evcc-io/evcc/pull/6267/files#diff-c456526d4bd9022a488ce32de208b40d86016e2c40b0c81a2dffe557c186377dL1337